src/decode_all.rs

```diff
@@ -23,15 +23,14 @@ pub trait DecodeAll: Sized {
 	/// Decode `Self` and consume all of the given input data.
 	///
 	/// If not all data is consumed, an error is returned.
-	fn decode_all(input: &[u8]) -> Result<Self, Error>;
+	fn decode_all<I: Input>(input: &mut I) -> Result<Self, Error>;
```

Comment:
Please just keep `[u8]`, because `Input` doesn't give you any guarantee that it can return a `remaining_len`

Reply:
Or we document this. But then we also need to return the error of `remaining_len`. If remaining_len would return `None` we would need to try to read on byte to ensure that there isn't anything anymore.